eZ/Publish/Core/FieldType/BinaryBase/BinaryBaseStorage.php

```diff
@@ -72,31 +72,19 @@ public function storeFieldData(VersionInfo $versionInfo, Field $field, array $co
             return false;
         }
 
-        // no mimeType means we are dealing with an input, local file
-        if (!isset($field->value->externalData['mimeType'])) {
-            $field->value->externalData['mimeType'] =
-                $this->mimeTypeDetector->getFromPath($field->value->externalData['inputUri']);
-        }
-
-        $storedValue = $field->value->externalData;
-
-        // The file referenced in externalData MAY be an existing IOService file which we can use
-        if ($storedValue['id'] === null) {
-            $createStruct = $this->IOService->newBinaryCreateStructFromLocalFile(
-                $storedValue['inputUri']
-            );
-            $storagePath = $this->pathGenerator->getStoragePathForField($field, $versionInfo);
-            $createStruct->id = $storagePath;
+        if (isset($field->value->externalData['inputUri'])) {
+            $field->value->externalData['mimeType'] = $this->mimeTypeDetector->getFromPath($field->value->externalData['inputUri']);
```

Comment:
q: why are you using $this->mimeTypeDetector to calculate the mimeType instead of just getting it from $createStruct ? You overwrite it anyway $field->value->externalData['mimeType'] on that line...